src/arch/mips.py

```diff
@@ -47,16 +50,36 @@ def translate(self, input: IO[str]) -> Functions:
 
                     vals = [
                             Literal((opcode << 26) | (fcnt & 0x3f), TYPE_U32),
-                            Binary(OP_SHL, Var('rs'), 21),
-                            Binary(OP_SHL, Var('rt'), 16),
-                            Binary(OP_SHL, Var('rd'), 11),
-                            Binary(OP_SHL, Var('shift'), 6)
+                            Binary(OP_SHL, cast(Var('rs'), 5), 21),
+                            Binary(OP_SHL, cast(Var('rt'), 5), 16),
+                            Binary(OP_SHL, cast(Var('rd'), 5), 11),
+                            Binary(OP_SHL, cast(Var('shift'), 5), 6)
                     ]
 
                     func += Set(TYPE_U32, reduce(lambda a, b: Binary(OP_BITWISE_OR, a, b), vals))
 
                     yield func
                 
+                elif mode == 'RI':
+                    # type RI
+                    # mode for branches
+                    # (opcode: 6b) (register source: 5b) (funct: 5b) (imm: 16b)
+                    func = Function(name, [ param('rs', TYPE_MIPS_REG, TYPE_U32),
+                                            param('target', TYPE_U16, TYPE_U32)])
```

Comment:
I'mma add you to add a single space between the first `)` and the following `]` at the end of the line, for clarity